internal/xcuitest/config.go

```diff
@@ -226,3 +229,49 @@ func SortByHistory(suites []Suite, history insights.JobHistory) []Suite {
 	}
 	return res
 }
+
+// ShardSuites applies sharding by provided shard config
+func ShardSuites(p *Project) error {
+	var suites []Suite
+	for _, s := range p.Suites {
+		if s.ShardConfig == "" {
+			suites = append(suites, s)
+			continue
+		}
+		splitedSuites, err := parseShardConfig(s, p.Sauce.Concurrency)
+		if err != nil {
+			log.Warn().Err(err).Msgf("failed to get tests from %s, use default settings", s.ShardConfig)
```

Comment:
Log msg should start with a capital letter (not to be confused with error messages you are writing as part of an `error` you want to return).
```suggestion
			log.Warn().Err(err).Msgf("Failed to get tests from %q. Using default settings.", s.ShardConfig)
```

Reply:
Having said that, I think we should fail and not log a warning. As a user, I'd expect a quick feedback loop if I messed up. I'd rather not wait for all jobs to finish (or have to cancel mid-flight) so I can fix the sharding configuration.